types/requestidlecallback/index.d.ts

```diff
@@ -15,13 +15,11 @@ interface IdleDeadline {
 }
 
 interface IdleRequestOptions {
-    timeout: number;
+    timeout?: number;
 }
 
 type IdleCallbackHandle = number;
 
-type IdleRequestCallback = (deadline: IdleDeadline) => void;
```

Comment:
@sandersn 

What are you doing?

Reply:
I am pretty sure in TypeScript 4.4 `IdleRequestCallback` is not added, not even mentioning TypeScript 4.4 hasn't been released yet.

Reply:
It is added here: https://github.com/microsoft/TypeScript/pull/44684/files#diff-dc0eab937d15e62545da3ed7b4f40ad6b24f15dd88fbc6ceda2bfb4ed8356eb0R17905

4.4's beta ships this week, but packages on Definitely Typed have to compile with the nightly version of Typescript. requestidlecallback polyfills types from the DOM, so it has to match them exactly. Unfortunately, type aliases can only be defined once, so requestidlecallback can't declare IdleRequestCallback. However, older Typescripts could at continue using `request` and `requestIdleCallback` and others if it were inlined here.

I did that in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54166

Reply:
> 4.4's beta ships this week, but packages on Definitely Typed have to compile with the nightly version of Typescript. requestidlecallback polyfills types from the DOM, so it has to match them exactly. Unfortunately, type aliases can only be defined once, so requestidlecallback can't declare IdleRequestCallback. However, older Typescripts could at continue using `request` and `requestIdleCallback` and others if it were inlined here.

Then it should be published in a major version instead of a minor version.  Don't you know you are breaking all prior TS 4.4 users?